pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
```

Comment:
Do we need a condition to express (permanent) failure?

Reply:
I see for SecurityPolicy CR, it only has "Ready" condition with values: true/false/unknown.  Could we also use the same way for all other CRs?

Reply:
In our implementation, do we have there different cases: 1) CRD validation or NSX-T API validation failed (and we will not retry for these cases); 2) NSX-T realization failed, but we are retrying; 3) NSX-T realization failed and we stop re-trying?

And "Ready" and "NotReady" are just opposites. Why we need both? Maybe "NotReady" actually means "Failure" (and we no longer re-try)?

Reply:
I think the 3 cases above are all there.
But I think we can only have "Ready" condition type in common file https://github.com/vmware-tanzu/nsx-operator/pull/108/, the value can be True/False/Unknown.  Do you think it is OK? @jianjuns 
For specific CRDs, they can define their own ConditionType if needed in their _types.go.

Reply:
Ok let us discuss there.